### PR TITLE
feat: add extra args param to external-dns deployment

### PR DESCRIPTION
**Pull request by randalljohnson**
Allow an `extraArgs` parameter to be specified which passes additional cli flags to the external-dns binary.

### CODE REVIEW

File: .codegen/certificates/cert-manager/deployment.yaml [hunk 1]
```diff
@@ -136,6 +136,7 @@ spec:
             - --v=2
             - --cluster-resource-namespace=$(POD_NAMESPACE)
             - --leader-election-namespace=$(POD_NAMESPACE)
+            - --issuer-ambient-credentials
```
Comment (randalljohnson) [inline .codegen/certificates/cert-manager/deployment.yaml:139]: This is required for cert-manager to work with `Issuer` types and not just `ClusterIssuer`

File: .codegen/gateway/external-dns/rbac.yaml [hunk 2]
```diff
@@ -56,6 +56,22 @@ rules:
...
+      - watch
+      - list
+  - apiGroups:
+      - gateway.solo.io
+    resources:
+      - virtualservices
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (randalljohnson) [inline .codegen/gateway/external-dns/rbac.yaml:74]: Allows external-dns to monitor gloo resources.

File: .codegen/gateway/gloo-edge/settings.yaml [hunk 3]
```diff
@@ -20,13 +20,13 @@ spec:
...
     disableProxyGarbageCollection: false
-  discoveryNamespace: nukleros-gateway-system
+  discoveryNamespace: nukleros-gateway-system # +operator-builder:field:name=namespace,default="nukleros-gateway-system",type=string
   kubernetesArtifactSource: {}
   kubernetesConfigSource: {}
   kubernetesSecretSource: {}
   refreshRate: 60s
-
   gateway:
+    persistProxySpec: true
```
Comment (randalljohnson) [inline .codegen/gateway/gloo-edge/settings.yaml:29]: This config setting is required to have Gloo persist its internal `Proxy` objects to the etcd custom resource. This is what's consumed by external dns to map hostnames to lb endpoints.

File: apis/gateway/v1alpha1/glooedge/mutate/gateway_namespace_gateway_proxy.go [hunk 4]
```diff
@@ -36,7 +40,37 @@ func MutateGatewayNamespaceGatewayProxy(
...
+		spec["ssl"] = portSpec.SSL
+
+		if err := unstructured.SetNestedMap(target.Object, spec, "spec"); err != nil {
+			return mutatedGateways, fmt.Errorf("failed to set spec on gateway: %w", err)
+		}
+
+		target.SetName(portSpec.Name)
+
+		mutatedGateways = append(mutatedGateways, target)
+	}
```
Comment (randalljohnson) [inline apis/gateway/v1alpha1/glooedge/mutate/gateway_namespace_gateway_proxy.go:73]: This logic was in the wrong function, so it was never being called. This is me moving it to the correct function.

**Review by lander2k2 — APPROVED** [review]

**Comment by lander2k2** [comment]
@randalljohnson PR approved but I have one question that's not too important.  Should we update the sample manifest in `config/samples/` to include the use of that new `extraArgs` field?  If you think so, feel free to add a commit for that before merging.

**Comment by randalljohnson** [comment]
> @randalljohnson PR approved but I have one question that's not too important. Should we update the sample manifest in `config/samples/` to include the use of that new `extraArgs` field? If you think so, feel free to add a commit for that before merging.

Good call, updated that sample and Makefile to include an example.

**Pull request merged**